src/scrub.S

```diff
@@ -42,81 +81,37 @@ __metal_before_start:
 
     la      t0, __metal_boot_hart
     csrr    a5, mhartid
+
     /* Disable machine interrupts to be safe */
```

Comment:
I think this operation should be included in metal_memory_scrub so that function will operate correctly
if called outside of this environment. What other settings do we need to ensure that metal_memory_scrub does not cause a fault if run later in the program?

Reply:
Out of reset, bus error interrupts are masked out, with interrupts disabled. Once user enables BEU, registers a corresponding isr, it will catch ecc errors as they occur.
metal_memory_scrub now disables / restores global interrupts, which should avoid any spurious ecc error leading to bus error interrupt. It limits this function to be called only from m-mode. (writing to mstatus in u/s mode will raise exception).